### ISSUE TITLE
Maya 2023

### ISSUE THREAD

**Issue by PierreUr** [jhoolmans/mayaImporterBVH#6]
Install on Scriptfolder

Start with Code:

import bvh_importer
bvh_importer.BVHImporterDialog()

get Error:

import bvh_importer

bvh_importer.BVHImporterDialog()
# Error: SyntaxError: file <maya console> line 1: invalid syntax

**Comment by yoponchik** [comment]
Same here; I get '# Error: SyntaxError: file <maya console> line 1: invalid syntax'

**Comment by nekavoq** [comment]
For anyone having this issue, there are 2 main problems with the newer Maya versions:

1.  They have permanently shifted from Python 2 to Python 3, making it incredibly difficult (if not, impossible) to use old python scripts (RIP Nightshade UV you were the GOAT)
2. Believe it or not, pymel isn't installed by default, you have to manually get it yourself

I'm not a programmer, so I thought alright, GUI doesn't even pop up, syntax error, maybe it's the Python version thing. I paste the code into an online convertor to convert it to Python 3, and try to run the script. Doesn't work, can't find pymel, [turns out you have to get it yourself](https://forums.autodesk.com/t5/maya-programming/maya-2022-pymel-how-to-import/td-p/10409310); ok cool. Pymel is in, let's run the script, success! The GUI pops up!

But the script doesn't work.

`# Error: AttributeError: file C:\Users/User/Documents/maya/2024/scripts\bvh_importer.py line 187: '_io.TextIOWrapper' object has no attribute 'next'`

And at this point I have no idea how to go about it anymore; tried my best but this is far beyond what I can try to figure out.

Hoping someone updates this to work with newer Maya, in the mean time, you're gonna have to dig out an older version of Maya to work with older py scripts. I'm using 2020.

tl;dr, use an older Maya, new Maya changes makes it impossible to run old py scripts; alternatively someone amends the script to work on new Maya

**Comment by xiaoyang333** [comment]
next（）用readline（）代替就行了

**Comment by nekavoq** [comment]
Thank you xiaoyang333!

Replacing "next" with "readline" on line 187 does indeed fix the issue!

To recap:

1. Get the script as usual and put it in your Maya/20XX/scripts folder
2. Edit "next" to "readline" in line 187
3. Make sure you have pymel installed
4. In Maya, using Python run this code
```
import bvh_importer
bvh_importer.BVHImporterDialog()
```
5. You can now import BVH files!

Again, thank you xiaoyang333!

**Comment by jhoolmans** [comment]
Thanks everyone for trying and pointing out the issue.

I haven't been following the script since a long time but I will try to keep it more up to date.

The latest update should support newer Maya versions while the older version of the script is still available in the [python2 branch](https://github.com/jhoolmans/mayaImporterBVH/tree/python2).

Let me know if this update fixes the issues you've been having. 👍

**Comment by nekavoq** [comment]
I can confirm that the new update works on Maya 2024!
